sources/packages/api/src/services/application/application.service.ts

```diff
@@ -53,4 +58,42 @@ export class ApplicationService extends RecordDataModelService<Application> {
       .getOne();
     return application;
   }
+
+  async updateAssessmentInApplication(
+    applicationId: number,
+    application: GetApplicationDataDto,
+    assessment: ApplicationAssessmentDTO,
+  ): Promise<UpdateResult> {
+    const populatedApplication = await this.populateApplication(
+      application,
+      assessment,
+    );
+    return this.repo.update(applicationId, populatedApplication);
```

Comment:
The code doesn't need all the applications from the database to perform an update.
Please consider performing the update in some way similar to the below.
```
return this.repo.update(applicationId, { assessment });
```